app/frontend/components/pages/delegations/delegationHistoryPage.tsx

```diff
@@ -0,0 +1,145 @@
+  return (
+    <li className="delegations-history-item">
+      <div className="label">Stake Delegation</div>
+      <EpochDateTime epoch={stakeDelegation.epoch} dateTime={stakeDelegation.dateTime} />
+      <div>
+        New pool: <span className="bold">{stakeDelegation.newStakePool.name}</span>
+        <CopyPoolId value={stakeDelegation.newStakePool.id} />
+      </div>
+      {stakeDelegation.oldStakePool ? (
+        <div>
+          Previous pool: {stakeDelegation.oldStakePool.name}
+          <CopyPoolId value={stakeDelegation.oldStakePool.id} />
+        </div>
+      ) : (
+        ''
+      )}
+    </li>
+  )
+}
+
+export interface StakingReward extends DelegetionHistoryObject {
+  reward: number
+  stakePool: StakePool
+}
+
+const StakingRewardItem = ({stakingReward}: {stakingReward: StakingReward}) => {
+  return (
+    <li className="delegations-history-item">
+      <div>
+        <div className="label">Staking reward</div>
+        <EpochDateTime epoch={stakingReward.epoch} dateTime={stakingReward.dateTime} />
+      </div>
+      <div>
+        <div>Reward: {stakingReward.reward}</div>
+        <div className="grey">
+          {stakingReward.stakePool.name}
+          <CopyPoolId value={stakingReward.stakePool.id} />
+        </div>
+      </div>
+    </li>
+  )
+}
+
+export interface RewardWithdrawal extends DelegetionHistoryObject {
+  credit: number
+}
+
+const RewardWithdrawalItem = ({rewardWithdrawal}: {rewardWithdrawal: RewardWithdrawal}) => {
+  return (
+    <li className="delegations-history-item">
+      <div className="space-between">
+        <div>
+          <div className="label">Reward withdrawal</div>
+          <EpochDateTime epoch={rewardWithdrawal.epoch} dateTime={rewardWithdrawal.dateTime} />
+        </div>
+        <div>
+          <div className="transaction-amount credit">{rewardWithdrawal.credit}</div>
+        </div>
+      </div>
+    </li>
+  )
+}
+
+interface Props {
+  delegationHistory: any
+}
+
+const DelegationHistoryObjectToItem = {
+  [DelegetionHistoryItemType.StakeDelegation]: (x: DelegetionHistoryObject) => (
+    <StakeDelegationItem stakeDelegation={x as StakeDelegation} />
+  ),
+  [DelegetionHistoryItemType.StakingReward]: (x: DelegetionHistoryObject) => (
+    <StakingRewardItem stakingReward={x as StakingReward} />
+  ),
+  [DelegetionHistoryItemType.RewardWithdrawal]: (x: DelegetionHistoryObject) => (
+    <RewardWithdrawalItem rewardWithdrawal={x as RewardWithdrawal} />
+  ),
+}
+
+class DelegationHistoryPage extends Component<Props> {
+  render({delegationHistory}) {
+    const items = delegationHistory.map((data: DelegetionHistoryObject) => {
+      try {
+        return DelegationHistoryObjectToItem[data.type](data)
+      } catch (e) {
+        return ''
+      }
+    })
+
+    return (
+      <div className="delegations-history card">
+        <h2 className="card-title">Staking History</h2>
+        {delegationHistory.length === 0 ? (
```

Comment:
This could be simplified to just `!delegationHistory.length`

Reply:
At least to me, the original code seems more readable to me, it's also used like that in transactions history

Reply:
I think its readable enough for a variable that is either zero or something else (like length),  but I dont care that much.